From d55d5f0323aec796f3457f5ef6642a78c0f7648b Mon Sep 17 00:00:00 2001
From: Rui Wang <ruwang@redhat.com>
Date: Thu, 21 Aug 2014 04:13:46 -0400
Subject: [acpi] x86: Fix extended error log driver to depend on CONFIG_X86_LOCAL_APIC

Message-id: <1408594432-28006-12-git-send-email-ruwang@redhat.com>
Patchwork-id: 88493
O-Subject: [PATCH RHEL7.1 BZ1076655 11/17] ACPI, x86: Fix extended error log driver to depend on CONFIG_X86_LOCAL_APIC
Bugzilla: 1076655
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "Chen, Gong" <gong.chen@linux.intel.com>

Backport of:
Original commit: 9ebddac7ea2a1f4b4ce3335a78312a58dfaadb4d
Original Author: "Luck, Tony" <tony.luck@intel.com>
Original Date: Fri, 8 Nov 2013 14:03:33 -0800

 Randconfig build by Fengguang's robot army reported:

    drivers/built-in.o: In function `extlog_print':
    >> acpi_extlog.c:(.text+0xcc719): undefined reference to `boot_cpu_physical_apicid'

 The config had CONFIG_SMP=n so we picked up this definition from:

   <asm/cpu.h>: #define cpu_physical_id(cpu) boot_cpu_physical_apicid

 But boot_cpu_physical_apicid is defined in arch/x86/kernel/apic/apic.c
 which is only built if CONFIG_X86_LOCAL_APIC=y.

Signed-off-by: Chen, Gong <gong.chen@linux.intel.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/Kconfig b/drivers/acpi/Kconfig
index aeb8248..1d1879d 100644
--- a/drivers/acpi/Kconfig
+++ b/drivers/acpi/Kconfig
@@ -398,7 +398,7 @@ source "drivers/acpi/apei/Kconfig"
 
 config ACPI_EXTLOG
  tristate "Extended Error Log support"
- depends on X86_MCE
+ depends on X86_MCE && X86_LOCAL_APIC
  select EFI
  select UEFI_CPER
  default n
-- 
1.7.1